src/uct/ib/rc/accel/rc_mlx5.inl

```diff
@@ -264,6 +264,14 @@ uct_rc_mlx5_common_post_send(uct_rc_mlx5_iface_common_t *iface, int qp_type,
     if (fm_ce_se & MLX5_WQE_CTRL_CQ_UPDATE) {
         txwq->sig_pi = txwq->prev_sw_pi;
     }
+
```

Comment:
what about to hide a check above (line n. 252):
```
#if HAVE_TL_DC
    if (qp_type == UCT_IB_QPT_DCI) {
        uct_ib_mlx5_set_dgram_seg((void*)(ctrl + 1), av, grh_av, qp_type);
    }
#endif
```
to
```
uct_ib_mlx5_set_dgram_seg((void*)(ctrl + 1), av, grh_av, qp_type);
```
and update `uct_ib_mlx5_set_dgram_seg` to handle RC QP:
```
static UCS_F_ALWAYS_INLINE void
uct_ib_mlx5_set_dgram_seg(struct mlx5_wqe_datagram_seg *seg,
                          uct_ib_mlx5_base_av_t *av, struct mlx5_grh_av *grh_av,
                          int qp_type)
{
#if HAVE_TL_DC
    if (qp_type == IBV_QPT_RC) {
        return;
    } else if (qp_type == UCT_IB_QPT_DCI) {
        mlx5_av_base(&seg->av)->key.dc_key     = htobe64(UCT_IB_KEY);
    } else if (qp_type == IBV_QPT_UD)
#endif
    {
        mlx5_av_base(&seg->av)->key.qkey.qkey  = htonl(UCT_IB_KEY);
    }

    ucs_assert((qp_type == IBV_QPT_UD)
#if HAVE_TL_DC
               || (qp_type == UCT_IB_QPT_DCI)
#endif
               );

    mlx5_av_base(&seg->av)->dqp_dct            = av->dqp_dct;
    ...
}
```

After this change:
- 1 branch for DC QP instead of 2 branches (checks for `qp_type == UCT_IB_QPT_DCI` only once)
- 1 branch for RC QP (the same as it was)
- 0 branches for UD QP instead of 1 as it was (if `HAVE_TL_DC` is `0`)

but (as per out offline discussion) it couldn't give us performance gains, because `qp_type` in `uct_rc_mlx5_common_post_send` is a constant value

Reply:
yes, originally I implemented this in a bit other way (returning `res` from `uct_rc_mlx5_common_post_send`), which gave ~5-6 ns perf degradation. With this approach perf is improved a bit, as branch should be optimized by compiler.